### PR TITLE
fixing issue#2131: same description for all photos in multiple uploads 

### LINKED ISSUE #2131: Multiple Uploads: All pictures are having same description
**Summary:**

I uploaded some photos via multiple uploads. Each one having different description. After Uploading all photos have the same description. The description they have is the description of the last photo.

**Steps to reproduce:**

Upload multiple photos via multiple uploads option all having different description.

**Device and Android version:**

Motorola Moto G5 plus API 24

**Commons app version:**

2.9-release prodRelease build

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by sp2710**
**Fixes #2131 Multiple Uploads: All pictures are having same description** 

**What changes did you make and why?**
Added a new setNewDescription() method to set description of the photo. Earlier when we were writing any description it was calling setDescriptionText method for all the photos in multiple uploads and all photos were having same description. Now setDescriptionText method is only called for the desired image and all images are not having the same description.

Tested on Motorola Moto G5 plus with API level 24

### CODE REVIEW

**Comment by sp2710** [comment]
@domdomegg Well could you please let me know whether the changes that i had made are appropriate for the issue or not
Since I am new to open source. It would be a great pleasure to hear from you.

**Comment by domdomegg** [comment]
@sp2710 Am having a look now. Usually it can take some time for pull requests to be reviewed, especially if they have to do with uploading stuff as need to thoroughly test haven't broken anything there + actually need to upload files rather than just view some part of the app.

So far your commits have been good, and your work is definitely appreciated :)

A change that will be needed is formatting, we (and many other open source projects) follow the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html). In some places whitespace needs to be added. First though I'll review fully to check this does fix the issue and doesn't create new ones.

**Comment by domdomegg** [comment]
**Tests performed**

Unit tests pass

Upload multiple files works on
`2.9.0-debug-pr-2132~42c530f4` on `Galaxy Nexus (emulator)` with API level `28`

This does solve the issue.

File: app/src/main/java/fr/free/nrw/commons/upload/DescriptionsAdapter.java [hunk 1]
```diff
@@ -174,7 +174,10 @@ public void init(int position) {
                     descItemEditText.setCompoundDrawablesWithIntrinsicBounds(null, null, null, null);
                 }
 
-                descItemEditText.addTextChangedListener(new AbstractTextWatcher(description::setDescriptionText));
+                descItemEditText.addTextChangedListener(new AbstractTextWatcher(descriptionText->{
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/upload/DescriptionsAdapter.java]:
```suggestion
                descItemEditText.addTextChangedListener(new AbstractTextWatcher(descriptionText -> {
```

File: app/src/main/java/fr/free/nrw/commons/upload/DescriptionsAdapter.java [hunk 2]
```diff
@@ -174,7 +174,10 @@ public void init(int position) {
...
+                    setNewDescription(description,descriptionText,position);
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/upload/DescriptionsAdapter.java]:
```suggestion
                    setNewDescription(description, descriptionText, position);
```

File: app/src/main/java/fr/free/nrw/commons/upload/DescriptionsAdapter.java [hunk 3]
```diff
@@ -254,6 +257,11 @@ private Drawable getInfoIcon() {
         }
     }
 
+    private void setNewDescription(Description description, String descriptionText, int position) {
+        description=descriptions.get(position-1);
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/upload/DescriptionsAdapter.java]:
```suggestion
        description = descriptions.get(position - 1);
```

**Comment by domdomegg** [comment]
I don't fully understand why this actually fixes the issue. I feel like this should be basically equivalent to what was already there. I'd like you either to explain what exactly wasn't working before that is now, or for someone else to review who does understand it.

**Comment by sp2710** [comment]
Actually when i put some log messages inside the description Text Changed Listener i saw that the **description** object for which the setDescriptionText was called was not always the same for which it was desired. I put both **description** and **descriptions** inside that method as logs and seen the logs. I saw that **description** object was not always reffering to the same **descriptions**

**Pull request merged**